pkg/controller/messaginginfra/broker/controller.go

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright 2020, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+package broker
+
+import (
+	"context"
+	"fmt"
+
+	logr "github.com/go-logr/logr"
+
+	v1beta2 "github.com/enmasseproject/enmasse/pkg/apis/enmasse/v1beta2"
+	"github.com/enmasseproject/enmasse/pkg/controller/messaginginfra/cert"
+	"github.com/enmasseproject/enmasse/pkg/controller/messaginginfra/common"
+	"github.com/enmasseproject/enmasse/pkg/util"
+	"github.com/enmasseproject/enmasse/pkg/util/install"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	resource "k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+type BrokerController struct {
+	client         client.Client
+	scheme         *runtime.Scheme
+	certController *cert.CertController
+}
+
+func NewBrokerController(client client.Client, scheme *runtime.Scheme, certController *cert.CertController) *BrokerController {
+	return &BrokerController{
+		client:         client,
+		scheme:         scheme,
+		certController: certController,
+	}
+}
+
+func getBrokerLabels(infra *v1beta2.MessagingInfra) map[string]string {
+	labels := make(map[string]string, 0)
+	labels["component"] = fmt.Sprintf("broker-%s", infra.Name)
+	labels["app"] = "enmasse"
+
+	return labels
+}
+
+/*
+ * Reconciles the broker instances for an instance of shared infrastructure.
+ *
+ * Each instance of a broker is created as a statefulset. If we want to support HA in the future, each statefulset can use replicas to configure HA.
+ */
+func (b *BrokerController) ReconcileBrokers(ctx context.Context, logger logr.Logger, infra *v1beta2.MessagingInfra) error {
+	setDefaultBrokerScalingStrategy(&infra.Spec.Broker)
+	logger.Info("Reconciling brokers", "broker", infra.Spec.Broker)
+
+	labels := getBrokerLabels(infra)
+
+	// Update broker condition
+	brokersCreated := infra.Status.GetMessagingInfraCondition(v1beta2.MessagingInfraBrokersCreated)
+	brokersCreated.SetStatus(corev1.ConditionTrue, "", "")
```

Comment:
Wouldn't this always reset the timestamp if the original status was "false"?

Reply:
So the timestamp should always be that of the 'initial' error?